container-search/src/main/java/com/yahoo/search/dispatch/LoadBalancer.java

```diff
@@ -201,13 +207,58 @@ private int nextScoreboardIndex(int current) {
     }
 
     static class AdaptiveScheduler implements GroupScheduler {
-
+        enum Type {TIME, REQUESTS}
         private final Random random;
         private final List<GroupStatus> scoreboard;
 
-        public AdaptiveScheduler(Random random, List<GroupStatus> scoreboard) {
+        private static double toDouble(Duration duration) {
+            return duration.toNanos()/1_000_000_000.0;
```

Comment:
```suggestion
            return duration.toSeconds();
```

Reply:
No can do, This only gives number of whole seconds. Needs exact time as a floating point number in seconds.